packages/drivers/odsp-driver/src/rateLimiter.ts

```diff
@@ -40,7 +39,8 @@ export class RateLimiter {
                 });
             };
             this.tasks.push(task);
-            process.nextTick(this.sched.bind(this));
+            // eslint-disable-next-line @typescript-eslint/no-floating-promises
+            Promise.resolve().then(async () => this.sched());
```

Comment:
Curious, why not run this.sched() right here? I.e. without any promises.

Reply:
So lets say the limit is 10. Now the 11th task coming will push the task in the queue and then if we just call this.sched, it 
will fail this condition "if (this.maxRequests > 0 && this.tasks.length > 0)" and then nothing would happen. So the previous any of the task needs to completed before 11th task gets scheduled.

Reply:
Condition will be exactly the same - no difference. The only difference is that - the .then has not been called yet to set continuation for result of acquire().
I've raised PR to simplify this logic and add UTs.